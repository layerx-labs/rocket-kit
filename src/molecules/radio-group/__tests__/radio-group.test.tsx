import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { RadioGroup } from '../../..';

const radioButtons = [
  { value: 'option_1', label: 'Option 1', disabled: false, checked: undefined },
  { value: 'option_2', label: 'Option 2', disabled: false, checked: undefined },
];

describe('Checkbox', () => {
  it('renders', () => {
    const { asFragment } = render(
      <RadioGroup group="test-group" options={radioButtons} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
