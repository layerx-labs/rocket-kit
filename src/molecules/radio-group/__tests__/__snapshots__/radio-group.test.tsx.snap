// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup renders 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  margin: 0;
  padding: 0;
  list-style: none;
}

.c0 li:not(:last-child) {
  margin-bottom: 0.625rem;
}

.c0 input[type='radio']:checked ~ .check {
  border-color: #2ba8b6;
}

.c0 input[type='radio']:checked ~ .check::before {
  background-color: #56cad7;
}

.c0 + span {
  margin-top: 0.625rem;
}

.c1 {
  --size: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  min-height: var(--size);
}

.c1 input[type='radio'] {
  position: absolute;
  visibility: hidden;
}

.c1 label {
  position: relative;
  padding-left: calc(var(--size) + 5px);
  line-height: var(--size);
  vertical-align: middle;
  cursor: pointer;
  z-index: 1;
}

.c1 .check {
  position: absolute;
  top: 0;
  border: 2px solid #f4f4f4;
  border-radius: 100%;
  width: var(--size);
  height: var(--size);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .check::before {
  content: '';
  display: block;
  border-radius: 100%;
  height: calc(var(--size) / 1.5);
  width: calc(var(--size) / 1.5);
}

.c1:hover .check {
  border-color: #939393;
}

<div
    class="radio-group"
  >
    <ul
      class="c0"
      type="column"
    >
      <li
        class="c1"
      >
        <input
          id="radio-test-group-0"
          name="test-group"
          type="radio"
          value="option_1"
        />
        <label
          for="radio-test-group-0"
        >
          Option 1
        </label>
        <div
          class="check"
        />
      </li>
      <li
        class="c1"
      >
        <input
          id="radio-test-group-1"
          name="test-group"
          type="radio"
          value="option_2"
        />
        <label
          for="radio-test-group-1"
        >
          Option 2
        </label>
        <div
          class="check"
        />
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
